Tasks/AzureAppServiceManage/appinsightsmanage.ts

```diff
@@ -0,0 +1,163 @@
+            await this.configureInstrumentationKey(appInsightsResource);
+            await this.configureAppServiceAlwaysOnProperty();
+            await this.configureAppInsightsWebTest(appInsightsResource, publishingProfile);
+            console.log(tl.loc("SuccessfullyConfiguredAppInsights"));
+        }
+    }
+
+    private async installApplicationInsightsExtension(publishingProfile) {
+        tl.debug('Retrieved publishing Profile');
+        var anyExtensionInstalled = await extensionManage.installExtensions(publishingProfile, [APPLICATION_INSIGHTS_EXTENSION_NAME], []);
+        
+        if(!anyExtensionInstalled) {
+            tl.debug('No new extension installed');
+        }
+    }
+
+    private async getApplicationInsightResource() {
+        var appInsightsResources =  await azureRESTUtils.getApplicationInsightsResources(this.endpoint, this.appInsightsResourceGroupName);
+        if(this.appInsightsResourceName != null) {
+            for(var appInsightResource of appInsightsResources) {
+                if(appInsightResource.name.toLowerCase() == this.appInsightsResourceName.toLowerCase()) {
+                    return appInsightResource;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private async linkAppInsightsWithAppService(appInsightsResource) {
+        var appInsightsResourceTags =  appInsightsResource.tags;
+        var isAppInsightsAlreadyConfigured = false;
+        var appInsightsLinkingTag = await this.getAppServiceHiddenLink();
+
+        for(var tagKey in appInsightsResourceTags) {
+            if(tagKey == appInsightsLinkingTag) {
+                isAppInsightsAlreadyConfigured =  true;
+                break;
+            }
+        }
+
+        if(isAppInsightsAlreadyConfigured) {
+            tl.debug("App service '" + this.webAppName + "' is already linked to app insights : " + appInsightsResource.name)
+        } else {
+            appInsightsResourceTags[appInsightsLinkingTag] = "Resource";
+            appInsightsResource = await azureRESTUtils.updateApplicationInsightsResource(this.endpoint, this.appInsightsResourceGroupName, appInsightsResource);
+        }
+
+        return appInsightsResource;
+    }
+
+    private async getAppServiceHiddenLink() {
+        var appServiceDetails = await azureRESTUtils.getAppServiceDetails(this.endpoint, this.resourceGroupName, this.webAppName, this.specifySlotFlag, this.slotName);
+        return "hidden-link:" + appServiceDetails.id;
+    }
+
+    private async configureInstrumentationKey(appInsightsResource) {
+        var appInsightsInstrumenationKey = appInsightsResource.properties.InstrumentationKey;
+        var appServiceAppSettings = await azureRESTUtils.getWebAppAppSettings(this.endpoint, this.webAppName, this.resourceGroupName, this.specifySlotFlag, this.slotName);
+
+        if(appInsightsInstrumenationKey && appServiceAppSettings) {
```

Comment:
If you don't get the instrumentation key or the app settings, are you supposed to throw?